src/main/java/data/exceptions/TaskManagerException.java

```diff
@@ -52,6 +52,17 @@ public static void checkIfDateInFormat(String date) throws TaskManagerException
         }
     }
 
+    /**
+     * need to debug this method
+     */
+    public static void checkIfTaskExistsInCurrentDate(List<Task> dayTasks, int index) throws TaskManagerException {
+        if (index <= 0) {
+            throw new TaskManagerException("Invalid task number. Please try again.");
+        } else if (index > dayTasks.size()) {
+            throw new TaskManagerException("Task number does not exist. Please try again.");
+        }
+    }
+
```

Comment:
Great handling of the case where the user might accidentally enter a wrong index!